core/src/main/scala/slog/macros/LoggerImpl.scala

```diff
@@ -0,0 +1,37 @@
+package slog.macros
+
+import slog.{LevelLogBuilder, LogBuilder}
+
+import scala.reflect.macros.blackbox
+
+object LoggerImpl {
+  def infoImpl[F[_]](c: blackbox.Context): c.Expr[LevelLogBuilder[F]] = {
+    impl[F](c)(c.universe.TermName("info"))
+  }
+
+  def debugImpl[F[_]](c: blackbox.Context): c.Expr[LevelLogBuilder[F]] = {
+    impl[F](c)(c.universe.TermName("debug"))
+  }
+
+  def traceImpl[F[_]](c: blackbox.Context): c.Expr[LevelLogBuilder[F]] = {
+    impl[F](c)(c.universe.TermName("trace"))
+  }
+
+  def warnImpl[F[_]](c: blackbox.Context): c.Expr[LevelLogBuilder[F]] = {
+    impl[F](c)(c.universe.TermName("warn"))
+  }
+
+  def errorImpl[F[_]](c: blackbox.Context): c.Expr[LevelLogBuilder[F]] = {
+    impl[F](c)(c.universe.TermName("error"))
+  }
+
+  private def impl[F[_]](
+      c: blackbox.Context
+  )(level: c.universe.TermName): c.Expr[LevelLogBuilder[F]] = {
+    import c.universe._
+    val file = Literal(Constant(c.enclosingPosition.source.path.toString))
+    val line = Literal(Constant(c.enclosingPosition.line))
+    val tree = q"${c.prefix}.underlying.$level($file, $line)"
```

Comment:
I strongly against this. This is the exact opposite of what I had in mind when I stated the requirement that file name and line number should be in the log (and maybe also other things, line method and class names). We should delegate that to the underlying `slf4j` (or `Logback`)

Reply:
I'm afraid that it is not feasible in a maintainable way. That would require every logger implementation to be macro based, which is in my perspective not a good idea.

Reply:
I've said it before. Even though I understand Ondra's point I do not agree with it in this particular case. Moreover, it does not seem much important, it's just two lines of code which can be removed if we find out during implementation and first alpha releases that it shouldn't be there or the other way around. For now I would leave it and let's see.

Reply:
This is not just "two lines of code". It's more fundamental issue. What @sideeffffect proposes is essentially getting rid of traits and use single class implementation only. I think that this is not really a way forward as it would would make the library not so useful and hardly extensible.